apps/penxle.com/src/routes/(default)/[space]/Post.svelte

```diff
@@ -276,11 +262,13 @@
     const content = fragment.toJSON() as JSONContent[];
     return content;
   }
+
+  $: croppedThumbnail = $postRevision.croppedThumbnail as FragmentType<Image_image>;
```

Comment:
```suggestion
  let croppedThumbnail: FragmentType<Image_image>;
  $: croppedThumbnail = $postRevision.croppedThumbnail;
```

P2: reactive variable의 경우 요렇게 타입 선언을 하는 것을 추천드립니다

Reply:
`as FragmentType<Image_image>`을 빼면 `$postRevision.croppedThumbnail`이 `undefined`일 수 있다는 타입 오류가 나네요... 

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/OIyHyV8T3If5OJv5ys3X/643ebff9-db50-4d0f-bd33-02376e4cbf91.png)

Reply:
`let croppedThumbnail: FragmentType<Image_image> | undefined` 로 하면 되지 않을가용 (실제로도 undefined일 수 있는거 아니에요?)

Reply:
사실 `croppedThumbnail`은 존재하지 않을 가능성이 있으니까 변수 선언을 ` | undefined | null`로 해봤는데 이젠 다른 오류가 나요...
```
Type '({ id: string; } & { ' $fragmentRefs'?: { Image_image: Image_image; } | undefined; }) | null | undefined' is not assignable to type 'FragmentType<Image_image> | null | undefined'.
  Type '{ id: string; } & { ' $fragmentRefs'?: { Image_image: Image_image; } | undefined; }' is not assignable to type 'FragmentType<Image_image> | null | undefined'.
    Type '{ id: string; } & { ' $fragmentRefs'?: { Image_image: Image_image; } | undefined; }' is missing the following properties from type 'FragmentType<Image_image>': url, width, height, placeholderts(2322)

let croppedThumbnail: FragmentType<Image_image> | null | undefined
```

Reply:
근데 일단 애초에 `croppedThumbnail`을 왜 `FragmentType<Image_image>` 로 캐스팅하는지도 잘 모르겠는데... `<Image>` 컴포넌트에 넘기는거도 아닌 거 같은데 그냥 `id`랑 `url` 가져오면 되는거 아닌가요

Reply:
캐스팅 없이 바로 프로퍼티 참조하려니까 
```
Property 'url' does not exist on type '{ id: string; } & { ' $fragmentRefs'?: { Image_image: Image_image; } | undefined; }'.ts(2339)
```
오류가 나서 나스님께 물어본 방식대로 짰었어요...

Reply:
그러니까 지금 `Post.svelte` 컴포넌트가 실제로 필요로 하는건 `croppedThumbnail`의 `url` 인건데 왜 graphql에서 `croppedThumbnail` 안의 `...Image_image` fragment를 가져오려고 하는지? `Image_image` fragment는 `<Image>` 컴포넌트에 해당 이미지를 넘겨주려고 할 때 필요한건데 지금 `<Image $image={croppedThumbnail} />` 같은 코드가 없잖아요